route/route_test.go

```diff
@@ -923,3 +924,201 @@ func TestProcessEventMetrics(t *testing.T) {
+	req.Header.Set("Content-Type", "application/x-msgpack")
+	req.Header.Set("X-Honeycomb-Team", "test-api-key")
+
+	// Set up mux variables for dataset extraction
+	req = mux.SetURLVars(req, map[string]string{"datasetName": "test-dataset"})
+
+	// Call router.batch directly
+	w := httptest.NewRecorder()
+	router.batch(w, req)
+	assert.Equal(t, http.StatusOK, w.Code)
+
+	var responses []*BatchResponse
+	err = json.Unmarshal(w.Body.Bytes(), &responses)
+	require.NoError(t, err)
+	assert.Len(t, responses, len(batchEvents))
+
+	// Verify all responses are successful
+	for i, resp := range responses {
+		assert.Equal(t, http.StatusAccepted, resp.Status, "Response %d should be accepted", i)
+		assert.Empty(t, resp.Error, "Response %d should have no error", i)
+	}
+
+	mockMetrics := router.Metrics.(*metrics.MockMetrics)
+	assert.Equal(t, 1, mockMetrics.CounterIncrements["incoming_router_batch"])
+	assert.Equal(t, 3, mockMetrics.CounterIncrements["incoming_router_batch_events"])
+
+	var spans []*types.Span
+	for len(spans) < len(batchEvents) {
+		select {
+		case span := <-router.Collector.(*collect.MockCollector).Spans:
+			spans = append(spans, span)
+		default:
+			// All the spans should be in the channel before batch() returns.
+			t.Fatal("didn't get enough spans")
+		}
+	}
+
+	assert.Len(t, spans, len(batchEvents))
+	for i, span := range spans {
+		assert.Equal(t, batchEvents[i].Data["trace.trace_id"], span.TraceID)
+		assert.Equal(t, uint(batchEvents[i].SampleRate), span.SampleRate)
+		assert.Equal(t, batchEvents[i].Data, span.Data)
+	}
+}
+
+// discardResponseWriter implements http.ResponseWriter that discards all writes
+type discardResponseWriter struct {
+	header http.Header
+}
+
+func (d *discardResponseWriter) Header() http.Header {
+	if d.header == nil {
+		d.header = make(http.Header)
+	}
+	return d.header
+}
+
+func (d *discardResponseWriter) Write(p []byte) (int, error) {
+	return len(p), nil // Discard all writes
+}
+
+func (d *discardResponseWriter) WriteHeader(statusCode int) {
+	// Discard status code
+}
+
+func BenchmarkRouterBatch(b *testing.B) {
+	router := newBatchRouter(b)
+	batchEvents := createBatchEvents()
+	batchMsgpack, err := msgpack.Marshal(batchEvents)
+	require.NoError(b, err)
+
+	mockCollector := router.Collector.(*collect.MockCollector)
+
+	// Create HTTP request directly without server
+	req, err := http.NewRequest("POST", "/1/batch/test-dataset", nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/x-msgpack")
+	req.Header.Set("X-Honeycomb-Team", "test-api-key")
+
+	// Set up mux variables for dataset extraction
+	req = mux.SetURLVars(req, map[string]string{"datasetName": "test-dataset"})
+
+	// Create reusable discard response writer
+	w := &discardResponseWriter{}
+
+	b.ResetTimer()
+	for b.Loop() {
+		req.Body = io.NopCloser(bytes.NewReader(batchMsgpack))
+
+		router.batch(w, req)
+
+		// Drain spans to prevent channel from filling up
+		for len(mockCollector.Spans) > 0 {
```

Comment:
Should this draining logic be outside of `b.Loop()` so that it's not included in the benchmark result?